src/firefly/js/ui/DateTimePickerField.jsx

```diff
@@ -80,94 +81,76 @@ function isSameTimes(time1, time2) {
      return compareMoments(m1, m2);
 }
 
-export class DateTimePicker extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.onClose = this.onClose.bind(this);
-        this.onSelectedDate = this.onSelectedDate.bind(this);
+export function DateTimePicker({showInput,openPicker, value='', onChange, onChangeOpenStatus,wrapperStyle,
+                                   inputStyle, groupKey, timeFieldKey}) {
 
-        const {value = ''} = props;
-        const timeMoment = aMoment(value).isValid() ? aMoment(value) : value;   // a moment or original string value
-        this.state={timeMoment};  // store valid moment or string
-    }
-
-    onClose() {
-        const {onChangeOpenStatus} = this.props;
+    const [timeMoment, setTimeMoment] = useState(aMoment(value).isValid() ? aMoment(value) : value);
```

Comment:
Convert this to a function:

```javascript
const [timeMoment, setTimeMoment] = useState(() => aMoment(value).isValid() ? aMoment(value) : value);
```
This reason is that react will only call the function the first time it renders. If you don't make a function than the work is done every time even though react only uses it the first time.
Any computation in useState initialization should be a function.